react-ui/src/ControlPanel.jsx

```diff
@@ -32,11 +30,18 @@ export class ControlPanel extends PureComponent {
         <div className="col-sm-3">
           Image filter:
           <Select
-            options={options}
-            value={makeOption(this.props.pictureFilter)}
+            options={this.props.pictureFilterList}
+            value={this.props.pictureFilter}
             menuPlacement="top"
             onChange={this.setPictureFilter}
           />
+          <input type="range"
+            min="0" max="1" step="0.01"
+            value={this.props.pictureFilterWeight}
+            onChange={this.setPictureFilterWeight}
+          />
+          <input type="file" id="new-filter-fileinput" onChange={this.newPictureFilterFromImage} style={{display:'none'}} accept=".png,.jpg,.jpeg"/>
+          <label htmlFor="new-filter-fileinput" className="button tertiary">+</label>
```

Comment:
Hmm. I'll likely:
- Add some text explaining what's going on
- Maybe use material icons here

Reply:
Alternatively add it via rendering a special button within the select.